addons/mail/static/src/core/common/thread_model.js

```diff
@@ -983,6 +983,17 @@ export class Thread extends Record {
         }
     }
 
+    replaceMessages(tmpMsg, message) {
+        // The message from other personas (not self) should not replace the tmpMsg
+        if (tmpMsg && tmpMsg.in(this.messages) && message.author.eq(this.store.self)) {
+            this.messages.splice(this.messages.indexOf(tmpMsg), 1, message);
+            return;
+        }
+        if (message.notIn(this.messages)) {
+            this.messages.push(message);
+        }
```

Comment:
The method name is not really reflecting reality, as this might also add the message if it can't be replaced.
It's more like smart add/replace.